soc/arm/st_stm32/common/soc_gpio.c

```diff
@@ -0,0 +1,272 @@
+			} else {
+				LL_GPIO_SetPinPull(gpio, pin, LL_GPIO_PULL_DOWN);
+			}
+		}
+
+	} else {
+		temp = conf & (STM32_CNF_OUT_1_MASK << STM32_CNF_OUT_1_SHIFT);
+		if (temp == STM32_CNF_GP_OUTPUT) {
+			LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_OUTPUT);
+		} else {
+			LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_ALTERNATE);
+		}
+
+		temp = conf & (STM32_CNF_OUT_0_MASK << STM32_CNF_OUT_0_SHIFT);
+		if (temp == STM32_CNF_PUSH_PULL) {
+			LL_GPIO_SetPinOutputType(gpio, pin, LL_GPIO_OUTPUT_PUSHPULL);
+		} else {
+			LL_GPIO_SetPinOutputType(gpio, pin, LL_GPIO_OUTPUT_OPENDRAIN);
+		}
+
+		temp = conf & (STM32_MODE_OSPEED_MASK << STM32_MODE_OSPEED_SHIFT);
+		if (temp == STM32_MODE_OUTPUT_MAX_2) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_LOW);
+		} else if (temp == STM32_MODE_OUTPUT_MAX_10) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_MEDIUM);
+		} else {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_HIGH);
+		}
+	}
+#else
+	pin = 1 << pin;
+
+	u32_t temp = conf & (STM32_MODER_MASK << STM32_MODER_SHIFT);
+	if (temp == STM32_MODER_OUTPUT_MODE) {
+		LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_OUTPUT);
+
+		temp = conf & (STM32_OTYPER_MASK << STM32_OTYPER_SHIFT);
+		if (temp == STM32_OTYPER_PUSH_PULL) {
+			LL_GPIO_SetPinOutputType(gpio, pin, LL_GPIO_OUTPUT_PUSHPULL);
+		} else {
+			LL_GPIO_SetPinOutputType(gpio, pin, LL_GPIO_OUTPUT_OPENDRAIN);
+		}
+
+		temp = conf & (STM32_OSPEEDR_MASK << STM32_OSPEEDR_SHIFT);
+		if (temp == STM32_OSPEEDR_LOW_SPEED) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_LOW);
+		} else if (temp == STM32_OSPEEDR_MEDIUM_SPEED) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_MEDIUM);
+		} else if (temp == STM32_OSPEEDR_HIGH_SPEED) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_HIGH);
+#if defined(CONFIG_SOC_SERIES_STM32F2X) || defined(CONFIG_SOC_SERIES_STM32F4X) || \
+	defined(CONFIG_SOC_SERIES_STM32F7X) || defined(CONFIG_SOC_SERIES_STM32L0X) || \
+	defined(CONFIG_SOC_SERIES_STM32L4X)
+		} else if (temp == STM32_OSPEEDR_VERY_HIGH_SPEED) {
+			LL_GPIO_SetPinSpeed(gpio, pin, LL_GPIO_SPEED_FREQ_VERY_HIGH);
+#endif
+		} else {
+			return -EINVAL;
+		}
+	} else if (temp == STM32_MODER_INPUT_MODE) {
+		LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_INPUT);
+
+		temp = conf & (STM32_PUPDR_MASK << STM32_PUPDR_SHIFT);
+		if (temp == STM32_PUPDR_PULL_UP) {
+			LL_GPIO_SetPinPull(gpio, pin, LL_GPIO_PULL_UP);
+		} else if (temp == STM32_PUPDR_PULL_DOWN) {
+			LL_GPIO_SetPinPull(gpio, pin, LL_GPIO_PULL_DOWN);
+		} else {
+			LL_GPIO_SetPinPull(gpio, pin, LL_GPIO_PULL_NO);
+		}
+	} else if (temp == STM32_MODER_ANALOG_MODE) {
+		LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_ANALOG);
+	} else {
+		LL_GPIO_SetPinMode(gpio, pin, LL_GPIO_MODE_ALTERNATE);
+
+		if (pin < 8) {
+			LL_GPIO_SetAFPin_0_7(gpio, pin, altf);
+		} else {
+			LL_GPIO_SetAFPin_8_15(gpio, pin, altf);
+		}
+	}
+#endif /* CONFIG_SOC_SERIES_STM32F1X */
+
+	return 0;
+}
+
+int stm32_gpio_set(u32_t *base, int pin, int value)
```

Comment:
Could be moved in stm32 gpio driver (merged in gpio_stm32_write)

Reply:
So I'm thinking all functions in soc_gpio.c file can be moved into gpio_stm32.c file. Should moving them into gpio_stm32.c be a separate commit ?